protocol/lib/json/json_test.go

```diff
@@ -0,0 +1,91 @@
+package json_test
+
+import (
+	"testing"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib/json"
+)
+
+func TestIsValidJSON(t *testing.T) {
+	type testCase struct {
+		name    string
+		input   string
+		wantErr bool
+	}
+
+	testCases := []testCase{
+		{
+			name:    "Basic JSON string",
+			input:   `{"key": "value"}`,
+			wantErr: false,
+		},
+		{
+			name:    "Invalid string",
+			input:   `{"key": "value`,
+			wantErr: true,
+		},
+		{
+			name:    "Nested well formed JSON object",
+			input:   `{"key": {"nestedKey": "nestedValue"}}`,
+			wantErr: false,
+		},
+		{
+			name:    "Typo in nested JSON object",
+			input:   `{"key": {"nestedKey": "nestedValue}`,
+			wantErr: true,
+		},
+		{
+			name: "Nested well formed JSON object with array",
+			input: `{
+				"exchanges": [
+				  {
+					"exchangeName": "Binance",
+					"ticker": "\"LINKUSDT\"",
+					"adjustByMarket": "USDT-USD"
+				  },
+				  {
+					"exchangeName": "Bitstamp",
+					"ticker": "LINK/USD"
+				  },
+				  {
+					"exchangeName": "Bybit",
+					"ticker": "LINKUSDT",
+					"adjustByMarket": "USDT-USD"
+				  }
+				]
+			}`,
+			wantErr: false,
+		},
+		{
+			name: "Typo in JSON object with array",
+			input: `{
+				"exchanges": [
+				  {
+					"exchangeName": "Binance",
+					"ticker": "\"LINKUSDT\"",
```

Comment:
suuuper nit, but: these nested quotes have been removed from Binance, so as an example of valid exchange config (minus a typo,) this is out of date.

Reply:
Ack. Also I got these from the [testutil/constants/genesis.go](https://github.com/dydxprotocol/v4-chain/blob/8900a3d2d518191b10649725064e25d4a420e5ea/protocol/testutil/constants/genesis.go#L931) file - should we update those?

Reply:
Yes, looks like that should be updated - thank for pointing it out!